app/controllers/chargeback_controller.rb

```diff
@@ -208,28 +228,48 @@ def cb_rate_edit
       if params[:typ] == "copy" # if tab was not changed
         session[:changed] = true
         @sb[:rate_details] = []
+        @sb[:num_tiers] = []
+        @sb[:tiers] = []
         rate = ChargebackRate.find(obj[0])
         @sb[:rate] = ChargebackRate.new
         @sb[:rate].description = "Copy of " + rate.description
         @sb[:rate].rate_type = rate.rate_type
-        rate_details = rate.chargeback_rate_details
+        rate_details = rate.chargeback_rate_details.to_a
+        rate_details.sort_by! { |rd| [rd.group.downcase, rd.description.downcase] }
         # Create new rate detail records for copied rate record
-        rate_details.each do |r|
+        rate_details.each_with_index do |r, i|
+          @sb[:tiers][i] = []
           detail = ChargebackRateDetail.new
           detail.description = r[:description]
           detail.source = r[:source]
-          detail.rate = r[:rate]
           detail.per_time = r[:per_time]
           detail.group = r[:group]
           detail.per_unit = r[:per_unit]
           detail.metric = r[:metric]
           detail.chargeback_rate_detail_measure_id = r[:chargeback_rate_detail_measure_id]
+          r.chargeback_tiers.each do |t|
```

Comment:
These single character block arguments are very hard to follow, let alone when there's a heavy nesting of iterators. I know you're trying to save space, but I believe clarity in using full names of the objects is much more important. Perhaps you could expand on these? This is true across this PR.